src/Vat.sol

```diff
@@ -217,23 +217,23 @@ contract Vat {
         // system is live
         require(live == 1, "Vat/not-live");
 
-        Urn memory urn = urns[i][u];
-        Ilk memory ilk = ilks[i];
+        uint256 rate_ = ilks[i].rate;
         // ilk has been initialised
-        require(ilk.rate != 0, "Vat/ilk-not-init");
+        require(rate_ != 0, "Vat/ilk-not-init");
 
+        Urn memory urn = urns[i][u];
```

Comment:
Does it make sense to add the urn to the stack vars to reduce mloads? `urn.art` and `urn.ink` are used frequently.

Reply:
We are right to the limit with variables, I'm pretty sure we would get a stack overflow. I could give a try though.

Reply:
Yeah separating them both to regular stack variables it hits stack too deep. I don't believe it is worth to keep seeing if there is a way to do so.